content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: 'Cyclic referential actions'
+metaTitle: 'Cyclic referential actions'
+metaDescription: 'Circular references can occur between models when each model references another creating a closed loop. Learn how Prisma identifies and suggests solutions to this potential problem.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+Circular references can occur between models when each model references another creating a closed loop. Most SQL databases handle these references out of the box. 
+
+With SQLServer however, if the [referential action](./) on the related field is set to something other than `NoAction`, the data model validator will throw an error.
+
+This is dependant on whether or not the scalar field is set to be optional or required. If no referential action is specified on the relation, the default values are used:
+
+| Clause     | Scalar field is optional | At least one scalar field is required |
+| :--------- | :----------------------- | :------------------------------------ |
+| `onDelete` | `SetNull`                | `NoAction`                            |
+| `onUpdate` | `Cascade`                | `Cascade`                             |
+
+</TopBlock>
+
+## Self relation
+
+The following model describes a self relation where the chicken references itself with fields that represent both the mother and father chickens.
+
+```prisma
+model Chicken {
+    id        Int      @id @default(autoincrement())
+    cock      Chicken? @relation(name: "a_self_relation")
+    hen       Chicken? @relation(name: "a_self_relation", fields: [chickenId], references: [id])
+    chickenId Int?
+}
+```
+
+This will result in the following error:
+
+```terminal
+Error parsing attribute "@relation": A self-relation must have `onDelete` and `onUpdate` referential actions set to `NoAction` in one of the @relation attributes. Implicit default `onDelete` and `onUpdate` values: `SetNull` and `Cascade`.
+```
+
+As the error message shows, the solution is to explicitly set the `onDelete` and `onUpdate` values to `NoAction` on one of the relations. This is because the relations are optional and the default referential action values are implicitly used.
```

Comment:
```suggestion
Since the default referential actions (`onDelete: SetNull`, `onUpdate: Cascade`) are creating a cycle, the solution is to explicitly set the `onDelete` and `onUpdate` values to `NoAction` on one of the relations.
```

Reply:
Proposing to simplify a bit and remind why the error occurs more explicitly

Reply:
Perfect. I debated including the last sentence. Should have gone with my gut, which said no 😅